File: auth-module-backend/routes/purchase.py
"""
充值中心路由模块
注册定价方案和兑换码相关路由
"""
from flask import Blueprint
from controllers.purchase_controller import get_pricing_plans, redeem_code
from middleware.auth_middleware import token_required

purchase_bp = Blueprint('purchase', __name__, url_prefix='/api/v1')

# 公开接口（无需认证）
purchase_bp.route('/pricing-plans', methods=['GET'])(get_pricing_plans)

# 需要登录的接口
purchase_bp.route('/purchase/redeem', methods=['POST'])(token_required(redeem_code))

File: ecom-ai-studio/src/api/purchase.ts
import { api } from './client'
import type { ApiResponse, PricingPlan } from '@/types'

/** 将后端蛇形命名定价方案转换为前端驼峰命名 */
function mapPricingPlan(raw: any): PricingPlan {
  return {
    id: raw.id,
    name: raw.name,
    price: Number(raw.price),
    coins: raw.coins,
    bonusCoins: raw.bonus_coins ?? raw.bonusCoins ?? 0,
    isActive: Boolean(raw.is_active ?? raw.isActive ?? false),
    sortOrder: raw.sort_order ?? raw.sortOrder ?? 0,
    createdAt: raw.created_at ?? raw.createdAt ?? '',
  }
}

export async function fetchPricingPlans(isActive?: boolean): Promise<PricingPlan[]> {
  const res = await api.get<ApiResponse<{ plans: any[] }>>('/pricing-plans', { isActive })
  return res.data.plans.map(mapPricingPlan)
}

export function redeemCode(data: {
  code: string
  walletType: string
  teamId?: string | null
}): Promise<
  ApiResponse<{
    coinsAdded: number
    newBalance: number
    codeInfo: { code: string; originalCoins: number }
  }>
> {
  return api.post<ApiResponse<any>>('/purchase/redeem', data)
}